tests/arch/arm64/arm64_svc_call/src/main.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Arm Limited (or its affiliates). All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#include "zephyr/arch/arm64/arm-smccc.h"
+
+/* SMC function IDs for Standard Service queries */
+#define ARM_STD_SVC_CALL_COUNT		0x8400ff00UL
+#define ARM_STD_SVC_VERSION		0x8400ff03UL
+#define ARM_STD_SVC_UNKNOWN		0xffffffffUL
+
+#define SMC_UNK				-1
+
+typedef void (*svc_call_method_t)(unsigned long, unsigned long,
+				  unsigned long, unsigned long,
+				  unsigned long, unsigned long,
+				  unsigned long, unsigned long,
+				  struct arm_smccc_res *);
+
+#ifdef CONFIG_SVC_CALL_USE_HVC
+	svc_call_method_t svc_call = arm_smccc_hvc;
```

Comment:
nit: shouldn't this be called `smc_call_method_t` ?

Reply:
That's right.